pkg/quota/v1/evaluator/core/resource_claims.go

```diff
@@ -99,10 +117,102 @@ func (p *claimEvaluator) MatchingResources(items []corev1.ResourceName) []corev1
 			strings.HasSuffix(string(item), corev1.ResourceClaimsPerClass /* by device class */) {
 			result = append(result, item)
 		}
+		if utilfeature.DefaultFeatureGate.Enabled(features.DRAExtendedResource) {
+			if strings.HasPrefix(string(item), corev1.ResourceImplicitExtendedClaimsPerClass /* by implicit extended resource name */) ||
+				isExtendedResourceNameForQuota(item) /* by extended resource name */ {
+				result = append(result, item)
+			}
+		}
 	}
 	return result
 }
 
+func (p *claimEvaluator) extendedResourceQuota(dcName string) corev1.ResourceName {
+	resource := corev1.ResourceName("")
+	if name, ok := p.deviceClassMapping.Get(dcName); ok {
+		resource = V1ExtendedResourceByDeviceClass(name)
+	}
+	return resource
+}
+
+func isPodRequest(name corev1.ResourceName, quantity resource.Quantity, reqs map[corev1.ResourceName]resource.Quantity) bool {
+	for r, q := range reqs {
+		if r == name && q.Cmp(quantity) == 0 {
+			return true
+		}
+	}
+	return false
+}
+
+func (p *claimEvaluator) setResourceQuantity(resourceMap map[corev1.ResourceName]resource.Quantity, quantity resource.Quantity, deviceClassName, name string, isExtendedResourceClaim bool, reqs map[corev1.ResourceName]resource.Quantity) {
+	deviceClassClaim := V1ResourceByDeviceClass(deviceClassName)
+	q := resourceMap[deviceClassClaim]
+	q.Add(quantity)
+	resourceMap[deviceClassClaim] = q
+	if !utilfeature.DefaultFeatureGate.Enabled(features.DRAExtendedResource) {
+		return
+	}
+	implicitExtendedResourceClaim := V1ImplicitExtendedResourceByDeviceClass(deviceClassName)
+	isImplicitExtendedResourceRequest := isPodRequest(implicitExtendedResourceClaim, quantity, reqs)
+	extendedResourceClaim := p.extendedResourceQuota(deviceClassName)
+	isExplicitExtendedResourceRequest := false
+	if extendedResourceClaim != "" && !isImplicitExtendedResourceRequest {
+		isExplicitExtendedResourceRequest = isPodRequest(extendedResourceClaim, quantity, reqs)
+	}
+	if !isExtendedResourceClaim || !isImplicitExtendedResourceRequest || isExplicitExtendedResourceRequest {
+		q := resourceMap[implicitExtendedResourceClaim]
+		q.Add(quantity)
+		resourceMap[implicitExtendedResourceClaim] = q
+	}
+	if extendedResourceClaim != "" {
+		if !isExtendedResourceClaim || isImplicitExtendedResourceRequest || !isExplicitExtendedResourceRequest {
+			q := resourceMap[extendedResourceClaim]
+			q.Add(quantity)
+			resourceMap[extendedResourceClaim] = q
+		}
+	}
+	if isImplicitExtendedResourceRequest {
+		delete(reqs, implicitExtendedResourceClaim)
+		return
+	}
+	if isExplicitExtendedResourceRequest {
+		delete(reqs, extendedResourceClaim)
+		return
+	}
+}
+
+func (p *claimEvaluator) verifyOwner(claim *resourceapi.ResourceClaim) (map[corev1.ResourceName]resource.Quantity, bool) {
+	if len(claim.OwnerReferences) == 0 {
```

Comment:
might be able to use the GetControllerOf helper to get the ownerReference with `controller:true`

if nil, return false

otherwise, verify that ownerReference instead of assuming [0]

can check Kind, APIVersion, lookup by namespace/name, then verify UID of the found pod matches

Reply:
I meant verify Kind/APIVersion in the ownerReference ... we know the pod is v1 because that's what we're watching ... we don't need to check APIVersion or Kind in the Pod object

Reply:
(also, unrelatedly, I just noticed the scheduler is setting BlockOwnerDeletion on the resourceClaim's pod ownerReference ... don't set that, we shouldn't block pod deletion on resourceclaim deletion ... can clean that up in a separate PR)

Reply:
Yes, i'm going to do that.